Runtime/Game/Requests/NotificationRequests.cs

```diff
@@ -0,0 +1,468 @@
+
+    /// <summary>
+    /// Enum of the different kinds of notification bodies possible, use this to figure out how to parse the notification body
+    /// </summary>
+    public enum LootLockerNotificationContentKind
+    {
+        group_reward = 0,
+        currency_reward = 1,
+        asset_reward = 2,
+        progression_reset_reward = 3,
+        progression_points_reward = 4,
+    };
+}
+
+namespace LootLocker.Requests
+{
+    //==================================================
+    // Data Definitions
+    //==================================================
+
+    /// <summary>
+    /// </summary>
+    public class LootLockerNotificationContextEntry
+    {
+        /// <summary>
+        /// The key for this context entry
+        /// </summary>
+        public string Key { get; set; }
+        /// <summary>
+        /// The value of this context entry
+        /// </summary>
+        public string Value { get; set; }
+    };
+
+    /// <summary>
+    /// </summary>
+    public class LootLockerNotificationRewardCurrency
+    {
+        /// <summary>
+        /// The date the Currency reward was created.
+        /// </summary>
+        public string Created_at { get; set; }
+        /// <summary>
+        /// The date the Currency reward was last updated.
+        /// </summary>
+        public string Updated_at { get; set; }
+        /// <summary>
+        /// The amount of Currency to be rewarded.
+        /// </summary>
+        public string Amount { get; set; }
+        /// <summary>
+        /// The details on the Currency.
+        /// </summary>
+        public LootLockerNotificationRewardCurrencyDetails Details { get; set; }
+        /// <summary>
+        /// The ID of a reward.
+        /// </summary>
+        public string Reward_id { get; set; }
+        /// <summary>
+        /// The ID of the Currency.
+        /// </summary>
+        public string Currency_id { get; set; }
+    }
+
+    /// <summary>
+    /// </summary>
+    public class LootLockerNotificationRewardCurrencyDetails
+    {
+        /// <summary>
+        /// The name of the Currency.
+        /// </summary>
+        public string Name { get; set; }
+        /// <summary>
+        /// The code of the Currency.
+        /// </summary>
+        public string Code { get; set; }
+        /// <summary>
+        /// The amount of the Currency.
+        /// </summary>
+        public string Amount { get; set; }
+        /// <summary>
+        /// The ID of the Currency.
+        /// </summary>
+        public string Id { get; set; }
+    }
+
+    /// <summary>
+    /// </summary>
+    public class LootLockerNotificationRewardProgression
+    {
+        /// <summary>
+        /// The date the Progression Points reward was created.
+        /// </summary>
+        public string Created_at { get; set; }
+        /// <summary>
+        /// The date the Progression Points was last updated.
+        /// </summary>
+        public string Updated_at { get; set; }
```

Comment:
Same comment about the rest of these, will not comment on all of them.

Reply:
Fixed